celery/backends/redis.py

```diff
@@ -102,9 +104,12 @@ def on_state_change(self, meta, message):
         self._maybe_cancel_ready_task(meta)
 
     def start(self, initial_task_id, **kwargs):
-        self._pubsub = self.backend.client.pubsub(
-            ignore_subscribe_messages=True,
-        )
+        if self._pubsub is None:
```

Comment:
@sihrc Re: @georgepsarakis's concern about setting threading.local() as a class attribute, you could assign self._thread to threading.local() inside start() if it is not yet an attribute.

Reply:
Thanks for the feedback @jamespeacock and also for testing this solution!